src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs

```diff
@@ -0,0 +1,1821 @@
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.ConstructorDeclaration);
+                    {
+                        N(SyntaxKind.FileKeyword);
+                        N(SyntaxKind.IdentifierToken, "record");
+                        N(SyntaxKind.ParameterList);
+                        {
+                            N(SyntaxKind.OpenParenToken);
+                            N(SyntaxKind.CloseParenToken);
+                        }
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void MethodNamedRecord_02_CSharp8()
+        {
+            UsingNode("""
+                class C
+                {
+                    file record() { }
+                }
+                """,
+                options: TestOptions.Regular8,
+                expectedBindingDiagnostics: new[]
+                {
+                    // (3,5): error CS0246: The type or namespace name 'file' could not be found (are you missing a using directive or an assembly reference?)
+                    //     file record() { }
+                    Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "file").WithArguments("file").WithLocation(3, 5),
+                    // (3,10): error CS0161: 'C.record()': not all code paths return a value
+                    //     file record() { }
+                    Diagnostic(ErrorCode.ERR_ReturnExpected, "record").WithArguments("C.record()").WithLocation(3, 10)
+                });
+
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.ClassDeclaration);
+                {
+                    N(SyntaxKind.ClassKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.MethodDeclaration);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "file");
+                        }
+                        N(SyntaxKind.IdentifierToken, "record");
+                        N(SyntaxKind.ParameterList);
+                        {
+                            N(SyntaxKind.OpenParenToken);
+                            N(SyntaxKind.CloseParenToken);
+                        }
+                        N(SyntaxKind.Block);
+                        {
+                            N(SyntaxKind.OpenBraceToken);
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void MethodNamedRecord_02_CSharpNext()
```

Comment:
This parse seems incorrect. I expect this to be a record with a missing name, not a constructor declaration(??)

Reply:
I think the parse is very similar as it is when the 'file' modifier is removed here. [SharpLab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDs4FsIIJYDGqe6MADjAgDQwhTIxUAmIA1AD4ACATAIwCwAKE4BmAAQ8xAYTEBvIWMVikBAPYImACgCUchUoC+Qg0A==).